src/software/restart_on_crash.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# The directory this script is in
+CURR_DIR=$(dirname -- "$(readlink -f -- "$BASH_SOURCE")")
+
+# the directory that contains the binary
+BIN_DIR=$(pwd)
+
+# The name of this script
+THIS_SCRIPT_FILENAME=$(basename "$0")
+
+# The number of arguments passed to this script
+NUM_ARGS=$#
+
+if [ $NUM_ARGS -ne 1 ]; then
+  echo "Error: Incorrect number of arguments to script"
+  echo "Usage: $THIS_SCRIPT_FILENAME [interface_name]"
+  echo "Choose from the following interfaces:"
+  ls /sys/class/net
+  exit 1
+fi
+
+if [[ "$1" == "help" ]]; then
```

Comment:
```suggestion
if [[ "$1" == "--help" ]]; then
```
--help seems more natural?